### PR TITLE
FEAT-#1871: introduce OmniSci based experimental engine

### LINKED ISSUE #1871: [OmniSci] Upstream OmniSci experimental engine to master branch

### PR DESCRIPTION
**Pull request by ienkovich**
## What do these changes do?

Technical PR to check production branch with experimental OmniSci engine

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1871  
- [x] tests passing

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,6 +13,7 @@ develop-eggs/
 dist/
 downloads/
 eggs/
+/data/
```
Comment (devin-petersohn) [inline .gitignore:16]: This seems specific to your development environment. Probably not needed here.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,13 @@
+“LEGAL NOTICE: Your use of this software and any required dependent software (the
```
Comment (devin-petersohn) [inline README.md:1]: Please remove product-specific language from PR

File: README.md [hunk 3]
```diff
@@ -9,7 +19,7 @@
 <a href="https://pypi.org/project/modin/"><img alt="" src="https://img.shields.io/badge/pypi-0.8.0-blue.svg" align="center"></a>
 </p>
 
-<p align="center"><b>To use Modin, replace the pandas import:</b></p>
+<p align="center"><b>To use Intel(r) Distribution of Modin, replace the pandas import:</b></p>
```
Comment (devin-petersohn) [inline README.md:22]: Please undo these changes

File: modin/backends/base/query_compiler.py [hunk 4]
```diff
@@ -45,6 +45,8 @@ def default_to_pandas(self, pandas_op, *args, **kwargs):
     # some of these abstract methods, but for the sake of generality they are
     # treated differently.
 
+    lazy_execution = False
```
Comment (devin-petersohn) [inline modin/backends/base/query_compiler.py:48]: If we are introducing this, it needs proper documentation.

File: modin/backends/base/query_compiler.py [hunk 5]
```diff
@@ -1182,3 +1203,7 @@ def delitem(self, key):
         return self.drop(columns=[key])
 
     # END __delitem__
+
+    @abc.abstractmethod
+    def has_multiindex(self):
```
Comment (devin-petersohn) [inline modin/backends/base/query_compiler.py:1208]: This will be good to put in a stand-alone PR. As a part of this change, we can change everything in the API layer where we are checking `isinstance(self.index, pandas.MultiIndex)`.

File: modin/backends/base/query_compiler.py [hunk 6]
```diff
@@ -122,6 +124,25 @@ def from_pandas(cls, df, data_cls):
         """
         pass
 
+    @classmethod
+    @abc.abstractmethod
+    def from_arrow(cls, at, data_cls):
```
Comment (devin-petersohn) [inline modin/backends/base/query_compiler.py:129]: This will be good to add in a separate stand-alone PR.

Comment (devin-petersohn) [reply modin/backends/base/query_compiler.py:129]: from and to arrow can be in the same PR.

File: modin/pandas/series.py [hunk 7]
```diff
@@ -92,7 +93,11 @@ def __init__(
                     )
                 )
             )._query_compiler
-        if len(query_compiler.columns) != 1 or (
+        if shape_hint == "column":
+            assert len(query_compiler.columns) == 1, "wrong shape hint"
+        elif shape_hint == "row":
```
Comment (devin-petersohn) [inline modin/pandas/series.py:98]: The goal is to avoid adding `shape_hint`.

This looks like the main issue for `shape_hint`, where we cannot determine whether or not to `transpose` because the logic for the `if len(...)` is in the API layer. Please correct me if I am wrong here.

The fix that makes the most sense to me is to add a call into the query compiler for something like `convert_to_columnar` (better name suggestion?) and then push this logic down.

Comment (ienkovich) [reply modin/pandas/series.py:98]: Yeah, this should help to move shape hint into the backend part, I'll try to move related logic into the query compiler.

**Comment by devin-petersohn** [comment]
@ienkovich can this be closed?

**Comment by ienkovich** [comment]
Yes, it has no functional differences now.
